client/src/components/CreateNewBoardModal.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { useContext, useRef, useEffect } from 'react';
+import styled from 'styled-components';
+import avatar from '../icons/avatar.jpg';
+import background from '../icons/teamBackground.jpg';
+import { DialogContext, ModalTypes } from '../contexts/DialogContext';
+
+const CreateNewBoardModal = () => {
+    const context = useContext(DialogContext);
+    const modalRef = useRef<HTMLDivElement>(null);
+
+    const onClickOutside = (e: any) => {
+        const element = e.target;
+        if (modalRef.current && !modalRef.current.contains(element)) {
+            e.preventDefault();
+            e.stopPropagation();
+            context.closeModalByType();
+        }
+    };
+
+    useEffect(() => {
+        document.body.addEventListener('click', onClickOutside);
+    });
+
+    return (
+        <Container>
+            <Modal ref={modalRef}>
+                <Header>Create board</Header>
+                <TypeWrapper>
+                    <TypeNav href="#">With team</TypeNav>
+                    <TypeNav href="#">With members</TypeNav>
+                </TypeWrapper>
+                <Input type="text" placeholder="Enter member's name" />
+                <TeamMember src={avatar} />
+                <TeamMember src={avatar} />
+                <TeamMember src={avatar} />
+                <Input type="text" placeholder="Title" />
+                <BackgroundLabel>Select background</BackgroundLabel>
+                <BackgroundContainer>
+                    <BackgroundItem src={background} alt="background" />
+                    <BackgroundItem src={background} alt="background" />
+                </BackgroundContainer>
+                <ButtonContainer>
+                    <CancelButton
+                        href="#"
```

Comment:
Don't use `href` unless it has a real url. If you want an `a` element to behave like a button, style it with css or just use a `button` element.